tests/bluetooth/host/id/bt_id_read_public_addr/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+FILE(GLOB SOURCES src/*.c)
+
+project(bt_id_read_public_addr)
+add_compile_definitions(test_unit_name="${PROJECT_NAME}")
+
+# Suppress the format-zero-length error if GNUC is used
+if (CMAKE_C_COMPILER_ID MATCHES "GNU")
+add_compile_options(-Wno-error=format-zero-length -Wno-format-zero-length)
+endif()
```

Comment:
Why is this needed? I don't think we should have this.

Reply:
Please have a look at this discussion https://github.com/zephyrproject-rtos/zephyr/pull/51329#discussion_r997767844

This is a workaround to the following warning `warning: zero-length gnu_printf format string [-Wformat-zero-length] ` generating while compiling `BT_DBG("");` for unit testing

Reply:
Something's not right. I believe the there is a bug in the logging subsystem. When the log it set to go to printf, the empty `BT_DBG("");` should print a newline. So the format-string should not be zero-length. We are hiding a bug by adding this compile option.

Reply:
Yes, from functional perspective, I'd expect that `BT_DBG("");` should print a new line, it's not the case.

Also, IMO for that particular case in `id.c`, it's not very helpful and I proposed to delete it, but I got some objections so I went for that workaround to unblock my PR.